helm/hpcc/templates/_warnings.tpl

```diff
@@ -0,0 +1,56 @@
+{{/*
+
+##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2022 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+##############################################################################
+
+*/}}
+{{/*
+Generate a list of warnings
+Pass in dict with root with warnings
+*/}}
+{{- define "hpcc.getWarnings" -}}
+ {{- $ctx := . -}}
+ {{- $warnings := list -}}
+ {{- $defaultVersion := .root.Values.global.image.version | default .root.Chart.Version -}}
+ {{- if (ne $defaultVersion .root.Chart.Version) -}}
+  {{- $warning := dict "source" "helm" "severity" "warning" -}}
+  {{- $_ := set $warning "msg" (printf "The expected image version for this helm chart is %s" .root.Chart.Version) -}}
+  {{- $_ := set $ctx "warnings" (append .warnings $warning) -}}
```

Comment:
although "." is ctx in this context - it is clearer if it explicitly references it:
```
{{- $_ := set $ctx "warnings" (append $ctx.warnings $warning) -}}
```